src/gs-shell.c

```diff
@@ -1931,14 +1974,80 @@ gs_shell_show_search (GsShell *shell, const gchar *search)
 			      (gpointer) search, TRUE);
 }
 
-void
-gs_shell_show_local_file (GsShell *shell, GFile *file)
+static void
+gs_shell_file_to_app_cb (GObject *source,
+			 GAsyncResult *res,
+			 gpointer user_data)
 {
-	g_autoptr(GsApp) app = gs_app_new (NULL);
+	GsPluginLoader *plugin_loader = GS_PLUGIN_LOADER (source);
+	GsShell *shell = GS_SHELL (user_data);
+	GsShellPrivate *priv = gs_shell_get_instance_private (shell);
+	g_autoptr(GError) error = NULL;
+	g_autofree gchar *tmp = NULL;
+	const gchar *removable_media_cat = NULL;
+	GsApp *app;
+
+	app = gs_plugin_loader_file_to_app_finish (plugin_loader, res, &error);
+	if (app == NULL) {
+		g_warning ("failed to convert to GsApp: %s", error->message);
+		gs_shell_set_mode (shell, GS_SHELL_MODE_OVERVIEW);
+		return;
+	}
+
+	/* print what we've got */
+	tmp = gs_app_to_string (app);
+	g_debug ("App from file:\n%s", tmp);
+
+	removable_media_cat = gs_app_get_metadata_item (app, "EndlessOS::RemovableMediaCategory");
+	if (removable_media_cat != NULL) {
+		GsCategory *category;
+		if (gs_app_get_metadata_item (app, "EndlessOS::ReloadOverview") == NULL) {
+			reload_overview_and_select_category (shell, removable_media_cat);
```

Comment:
Why do you need to reload the overview in this case? The metadata that calls for that was not set; perhaps you can just select the category.